library/src/scala/quoted/Type.scala

```diff
@@ -42,11 +51,20 @@ object Type:
    *  Returns None if the type is not a tuple singleton constant types.
    *
    *  Example usage:
-   *  ```scala
-   *  ... match
+   *  ```scala sc:nocompile
+   *  //{
+   *  import scala.deriving.*
+   *  def f(using Quotes) = {
+   *  import quotes.reflect.*
+   *  val expr: Expr[Any] = ???
+   *  //}
+   *  expr match {
    *    case '{ $mirrorExpr : Mirror.Sum { type MirroredElemLabels = label } } =>
    *      Type.valueOfTuple[label] // Option[Tuple]
    *  }
```

Comment:
@nicolasstucki
This snippet fails with `Type argument label does not conform to upper bound Tuple`.
Is this a bug?

Reply:
Yes, it is a bug.

```diff
-    case '{ $mirrorExpr : Mirror.Sum { type MirroredElemLabels = label } } =>
+    case '{ type label <: Tuple; $mirrorExpr : Mirror.Sum { type MirroredElemLabels = `label` } } =>

```

Reply:
Ok, changed here. Also created #13527